packages/Manager/src/main/resources/db/migration/V0041__observed_water_use_aggregated_daily.sql

```diff
@@ -0,0 +1,287 @@
+SET ROLE materialized_views_role;
+DROP MATERIALIZED VIEW IF EXISTS water_allocation_and_usage_by_area;
+RESET ROLE;
+
+DROP VIEW IF EXISTS observed_water_use_aggregated_daily CASCADE;
+
+CREATE OR REPLACE VIEW observed_water_use_aggregated_daily AS
+
+WITH filtered_obs AS (
+    SELECT osm.id,
+           osm.site_id,
+           os.name as site_name,
+           date(o.observed_at AT TIME ZONE 'NZST') AS day_observed_at,
+           osm.measurement_name,
+           o.amount
+    FROM observations o
+             LEFT JOIN observation_sites_measurements osm ON o.observation_measurement_id = osm.id
+             LEFT JOIN observation_sites os ON os.id = osm.site_id
+    WHERE osm.measurement_name IN ('Water Meter Reading', 'Water Meter Volume')
+      AND o.observed_at > '2022-11-01 00:00:00+12'
+),
+     first_values AS (
+         SELECT DISTINCT site_name, day_observed_at, measurement_name,
+                FIRST_VALUE(amount) OVER (PARTITION BY site_name, day_observed_at, measurement_name ORDER BY site_name, day_observed_at, measurement_name) as first_value
+         FROM filtered_obs
+         WHERE amount != 0
+     ),
+     last_values AS (
+         SELECT DISTINCT site_name, day_observed_at, measurement_name,
+                LAST_VALUE(amount) OVER (PARTITION BY site_name, day_observed_at, measurement_name ORDER BY site_name, day_observed_at, measurement_name ROWS BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING) as last_value
+         FROM filtered_obs
+         WHERE amount != 0
+     ),
+     daily_calculation AS (
+SELECT filtered_obs.site_name,
+       filtered_obs.day_observed_at,
+       filtered_obs.measurement_name,
+       CASE
+           WHEN filtered_obs.measurement_name = 'Water Meter Volume' THEN SUM(filtered_obs.amount)
+           WHEN count(*) = 1 AND LAG(max(filtered_obs.day_observed_at)) OVER (PARTITION BY id, filtered_obs.measurement_name ORDER BY id, filtered_obs.day_observed_at, filtered_obs.measurement_name) != filtered_obs.day_observed_at - interval '1 day' THEN NULL
+           WHEN count(*) = 1 THEN first_values.first_value - last_values.last_value
```

Comment:
this row (41) should remain what it was - the WHEN conditions get met when there is 1 reading per day, so it needs to take the 1 reading from that day and subtract the previous day to get usage for that day.